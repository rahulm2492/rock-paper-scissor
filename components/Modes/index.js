import React, { PropTypes } from 'react';
import styles from './Modes.module.scss';

const Modes = ({ currentMode, onToggleMode }) => {
	return(<div className={styles.Modes}>
		<span className="label">{currentMode?.label}</span><br />
		<button
			onClick={onToggleMode}
		>
			CHANGE MODE
		</button>
	</div>
)};


export default Modes;