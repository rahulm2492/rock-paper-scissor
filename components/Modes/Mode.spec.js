import { shallow, mount } from "enzyme";

import Modes from "./index";


/** @test {Mode Component} */
describe("Game Modes Component", () => {
  const func1=jest.fn();
  it("should render without crashing", () => {
    const wrapper = shallow(<Modes  />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should handle Mode of Game", () => {
    const wrapper = mount(<Modes onToggleMode={func1} currentMode={{label:'test'}}/>);
    
    const modeChangeButton = wrapper.find('button');
    modeChangeButton.simulate('click');
    expect(func1).toHaveBeenCalled();
   
    
});

});
