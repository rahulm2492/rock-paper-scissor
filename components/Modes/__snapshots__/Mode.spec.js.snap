// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Modes Component should render without crashing 1`] = `ShallowWrapper {}`;
